File: backend/src/main/java/com/electoral/entities/InstitucionEducativa.java
package com.electoral.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "instituciones_educativas")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InstitucionEducativa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String nombre;

    @Column(length = 500)
    private String direccion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parroquia_id", nullable = false)
    private Parroquia parroquia;

    @Column(length = 20)
    private String codigo;

    @Column(length = 50)
    private String tipo;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "institucion", cascade = CascadeType.ALL)
    private List<Recinto> recintos;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: conteo_electoral_app/lib/models/institucion_educativa.dart
class InstitucionEducativa {
  final int id;
  final String nombre;
  final int parroquiaId;
  final String? parroquiaNombre;
  final String? direccion;
  final String? codigo;
  final String? tipo;

  InstitucionEducativa({
    required this.id,
    required this.nombre,
    required this.parroquiaId,
    this.parroquiaNombre,
    this.direccion,
    this.codigo,
    this.tipo,
  });

  factory InstitucionEducativa.fromJson(Map<String, dynamic> json) {
    return InstitucionEducativa(
      id: json['id'],
      nombre: json['nombre'],
      parroquiaId: json['parroquiaId'],
      parroquiaNombre: json['parroquiaNombre'],
      direccion: json['direccion'],
      codigo: json['codigo'],
      tipo: json['tipo'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'parroquiaId': parroquiaId,
      'direccion': direccion,
      'codigo': codigo,
      'tipo': tipo,
    };
  }
}
